eng/native/configurecompiler.cmake

```diff
@@ -783,7 +855,16 @@ if (MSVC)
   # For Release builds, we shall dynamically link into uCRT [ucrtbase.dll] (which is pushed down as a Windows Update on downlevel OS) but
   # wont do the same for debug/checked builds since ucrtbased.dll is not redistributable and Debug/Checked builds are not
   # production-time scenarios.
-  set(CMAKE_MSVC_RUNTIME_LIBRARY MultiThreaded$<$<AND:$<OR:$<CONFIG:Debug>,$<CONFIG:Checked>>,$<NOT:$<BOOL:$<TARGET_PROPERTY:DAC_COMPONENT>>>>:Debug>)
+  set(CMAKE_MSVC_RUNTIME_LIBRARY MultiThreaded$<$<OR:$<CONFIG:Debug>,$<CONFIG:Checked>>:Debug>)
+
+  if (NOT CLR_CMAKE_ENABLE_SANITIZERS)
+    # Force uCRT to be dynamically linked for Release build
+    # We won't do this for sanitized builds as the dynamic CRT is not compatible with the static sanitizer runtime and
+    # the dynamic sanitizer runtime is not redistributable. Sanitized runtime builds are not production-time scenarios
```

Comment:
Technically we don't want to redistribute (ship) the sanitizer runtime on any platforms (we only want to copy it for tests). By "not redistributable" I mean it doesn't have the Microsoft Redistributable license and it's not meant to be shipped off the development machine (we end up doing it because we don't have VS build tools installed on our Helix test machines).